services/worker/src/worker/job_runners/split/descriptive_statistics.py

```diff
@@ -594,15 +578,59 @@ def _compute_statistics(
         )
 
     def compute_and_prepare_response(self, data: pl.DataFrame) -> StatisticsPerColumnItem:
-        stats = self._compute_statistics(data, self.name, self.n_samples, n_bins=self.n_bins)
+        stats = self._compute_statistics(data, column_name=self.name, n_samples=self.n_samples, n_bins=self.n_bins)
         return StatisticsPerColumnItem(
             column_name=self.name,
             column_type=ColumnType.LIST,
             column_statistics=stats,
         )
 
 
-SupportedColumns = Union[ClassLabelColumn, IntColumn, FloatColumn, StringColumn, BoolColumn, ListColumn]
+class AudioColumn(Column):
+    def __init__(self, *args: Any, n_bins: int, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self.n_bins = n_bins
+
+    @staticmethod
+    def get_duration(example: dict[str, Any]) -> float:
+        with io.BytesIO(example["bytes"]) as f:
+            return librosa.get_duration(path=f)  # type: ignore   # expects PathLike but BytesIO also works
+
+    @staticmethod
+    @raise_with_column_name
+    def _compute_statistics(
+        parquet_directory: Path,
+        column_name: str,
+        n_samples: int,
+        n_bins: int,
+    ) -> NumericalStatisticsItem:
+        table = pq.read_table(parquet_directory, columns=[column_name]).drop_null()
+        if table.shape[0] == 0:
+            return all_nan_statistics_item(n_samples)
+
+        data = table.to_pydict()[column_name]
```

Comment:
actually as all data is in-memory here, it might take quite a lot of additional RAM.
It will be up to 5Gb of RAM for parquet + same size to copy as python dict.
Should I load file by file instead of full directory at once?

Reply:
changed to reading file by file [1e1bf60](https://github.com/huggingface/datasets-server/pull/2612/commits/1e1bf6086c1d63d16f542fcd85d104c134f76161)